### DISCUSSION TITLE
How to invoke ContentField template for displaying from a ContentPart template?

### DISCUSSION THREAD

**Discussion by HengzheLi** [OrchardCMS/OrchardCore#9626] [Q&A]
I create a ContentPart with altering ContentFields to it. And there is a customize ContentPart template for the ContentPart's properties displaying. Then I want to invoke the ContentField's template from my customize ContentPart template. How to do it?

**Comment by ns8482e** [comment]
`ContentPart` uses composition.  All fields attached your content part are bydefault added to local `Content` zone. You can display that zone using  `@await DisplayAsync(Model.Content)` in your custom template.

Your custom ContentPart should render only Zones and you should not render your fields directly from your part's template instead you use placement rules.

e.g.  Instead of rendering all fields to default 'Content' zone you may want to have few fields to render on `Right`.
To achieve that you could define your template as `MyPart.cshtml`

```cshtml
<div>
    <div class="left">@await DisplayAsync(Model.Content)</div>
    <div> class="right>@await DisplayAsync(Model.Right)</div>
</div>
```

and use placement for your filed `MyField`
```json
"MyField" : {
    "place" : "Right:0"
}
```

**Reply by PiemP** [reply]
If I have created a DisplayDriver related to the ContentPart: how can I display the ContentFields shape?
